system/jlib/jtrace.cpp

```diff
@@ -107,6 +108,293 @@ class NoopSpanExporterFactory
+    {
+        sout << "{ ";
+        bool first = true;
+        for (const auto &kv : map)
+        {
+            if (!first)
+                sout << ",";
+            else
+                first = false;
+
+            sout << "\"" << kv.first << "\": \"";
+            opentelemetry::exporter::ostream_common::print_value(kv.second, sout);
+            sout << "\"";
+        }
+        sout << " }";
+    }
+
+    void printEvents(const std::vector<opentelemetry::sdk::trace::SpanDataEvent> &events)
+    {
+        sout << "{ ";
+        bool first = true;
+        for (const auto &event : events)
+        {
+            if (!first)
+                sout << ",";
+            else
+                first = false;
+
+            sout << "{ ";
+            sout << " \"Name\": " << event.GetName() << "\","
+                << " \"Timestamp\": " << event.GetTimestamp().time_since_epoch().count() << "\","
+                << " \"Attributes\": ";
+            printAttributes(event.GetAttributes());
+            sout << " }";
+        }
+
+        sout << " }";
+    }
+
+    void printLinks(const std::vector<opentelemetry::sdk::trace::SpanDataLink> &links)
+    {
+        bool first = true;
+
+        sout << "{ ";
+        for (const auto &link : links)
+        {
+            if (!first)
+                sout << ",";
+            else
+                first = false;
+
+            char trace_id[32] = {0};
+            char span_id[16]  = {0};
+            link.GetSpanContext().trace_id().ToLowerBase16(trace_id);
+            link.GetSpanContext().span_id().ToLowerBase16(span_id);
+            sout << " {"
+                << " \"TraceId\": \"" << std::string(trace_id, 32) << "\","
+                << " \"SpanId\": \"" << std::string(span_id, 16) << "\","
+                << " \"TraceState\": \"" << link.GetSpanContext().trace_state()->ToHeader() << "\","
+                << " \"Attributes\": ";
+            printAttributes(link.GetAttributes());
+            sout << " }";
+        }
+        sout << " }";
+    }
+
+    void printResources(const opentelemetry::sdk::resource::Resource &resources)
+    {
+        auto attributes = resources.GetAttributes();
+        if (attributes.size())
+            printAttributes(attributes);
+    }
+
+    void printInstrumentationScope(
+        const opentelemetry::sdk::instrumentationscope::InstrumentationScope &instrumentation_scope)
+    {
+        auto version = instrumentation_scope.GetVersion();
+        if (version.size())
+        {
+            sout << " \"" << instrumentation_scope.GetName() << "\": ";
+            sout << version;
+        }
+    }
+
+    bool shutDown = false;
+    std::ostringstream sout;
+    mutable opentelemetry::common::SpinLockMutex lock;
+
+    // Mapping status number to the string from api/include/opentelemetry/trace/canonical_code.h
+    std::map<int, std::string> statusMap{{0, "Unset"}, {1, "Ok"}, {2, "Error"}};
+};
+
+class JLogSpanExporterFactory
```

Comment:
This isn't wrong, but I think this is java style rather than normal c++ style to use a class with static members that provide the constuctors.  Much more common to call the constructor directly, have a normal function called createJLogSpanExporter(), or possibly use a static member of the class.